### PR TITLE
WIP: feature / Modal

### PR DESCRIPTION
**Pull request by bertya**
Work in progress. Open to review. Especially how what props it should have.

### CODE REVIEW

File: packages/modal/README.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+```jsx
+import Modal from 'pcln-modal'
+```
+
+```jsx
+<Modal
+  isOpen={this.state.isOpen}
+  onClose={handleClose}
+  bg="white"
+  title="Modal title"
```
Comment (jxnblk) [inline packages/modal/README.md]: This `title` prop looks like it should be `props.children` – might be worthwhile to expand on the README a bit more to lock down the API before jumping into the implementation

Comment (bertya) [reply packages/modal/README.md]: Yes, probably finding a chance with Wes and you to figure out what level of flexibility we want to give people and then knock down the API's

File: packages/modal/src/Modal.js [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+  disableCloseButton: PropTypes.bool,
+  enableOverflow: PropTypes.bool,
+  onClose: PropTypes.func,
+  bg: PropTypes.string,
+  zIndex: PropTypes.number,
+  title: PropTypes.string,
+  headerBg: PropTypes.string,
+  imgMode: PropTypes.bool,
+  className: PropTypes.string,
+  header: PropTypes.number
```
Comment (James300) [inline packages/modal/src/Modal.js]: Should header be `number`?

File: packages/modal/src/Modal.js [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+}) => (
+  <Transition
+    items={isOpen}
+    config={config.gentle}
+    from={{ opacity: 0, transform: 'scale(0.5)' }}
+    enter={{ opacity: 1, transform: 'scale(1)' }}
+    leave={{ opacity: 0, transform: 'scale(0.5)' }}
+  >
+    {isOpen =>
+      isOpen &&
```
Comment (rykimchi) [inline packages/modal/src/Modal.js]: If I'm interpreting this correctly, you can accomplish the same thing with the transition component. There is a property you can control with transition to accomplish the same thing. 

http://reactcommunity.org/react-transition-group/transition/#Transition-prop-unmountOnExit

I think we should allow the implementer to choose if the children should remain mounted or unmounted anyways.

File: packages/modal/README.md [hunk 4]
```diff
@@ -0,0 +1,24 @@
+
+# pcln-slider
```
Comment (BeniCheni) [inline packages/modal/README.md]: This seems to be `pcln-slider`?

File: packages/modal/src/Modal.js [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+                        id="pcln-modal-close"
+                        onClick={onClose}
+                        ml="auto"
+                      />
+                    )}
+                  </StyledHeader>
+                )}
+                {header !== 2 && !disableCloseButton && (
+                  <FloatCloseButton
+                    id="pcln-modal-close"
```
Comment (rykimchi) [inline packages/modal/src/Modal.js]: Is there a reason you need this to be an ID? 
We should avoid using ids to make these components as resuable as possible.

**Comment by BeniCheni** [comment]
Posted an issue to ask about `style.css` in [#102 in `reach-ui`](https://github.com/reach/reach-ui/issues/102).

File: packages/modal/src/Modal.js (hunk 5 above)
Comment (bertya) [reply packages/modal/src/Modal.js]: @rykimchi the ids are not used by the modal component itself at all. the purpose of ids is that the modal has so many layers, simply accepting className makes style change to deeper level hard, so these ids are helping people on that. Open to other suggestion.

**Comment by jxnblk** [comment]
Closing in favor of #448

**Pull request closed without merging**